// @evershop/evershop/src/modules/promotion/pages/admin/couponEdit+couponNew/FormContent.jsx
import PropTypes from 'prop-types';
import React from 'react';
import Area from '@components/common/Area';
import Button from '@components/common/form/Button';
import { useFormContext } from '@components/common/form/Form';
import { Card } from '@components/admin/cms/Card';
import './FormContent.scss';
import Layout from "@evershop/evershop/src/modules/cms/pages/frontStore/all/Layout";
import '../../../../i18n'
import {useTranslation} from "react-i18next";


export default function FormContent({ gridUrl }) {
  const {t, i18n} = useTranslation()
  const { state } = useFormContext();
  return (
      <>
        <div className="grid grid-cols-1 gap-2">
          <Card title= {t('general')}>
            <Card.Session>
              <Area id="couponEditGeneral" noOuter />
            </Card.Session>
          </Card>
          <Card title={t('discounttype')}>
            <Card.Session>
              <Area id="couponEditDiscountType" noOuter />
            </Card.Session>
          </Card>
          <div className="grid grid-cols-3 gap-x-2 grid-flow-row ">
            <div className="col-span-2 grid grid-cols-1 gap-2 auto-rows-max">
              <Card title={t('ordercontions')}>
                <Card.Session>
                  <Area id="couponEditLeft" noOuter className="col-8" />
                </Card.Session>
              </Card>
            </div>
            <div className="col-span-1 grid grid-cols-1 gap-2 auto-rows-max">
              <Card title={t('customercontions')}>
                <Card.Session>
                  <Area id="couponEditRight" className="col-4" noOuter />
                </Card.Session>
              </Card>
            </div>
          </div>
        </div>
        <div className="form-submit-button flex border-t border-divider mt-15 pt-15 justify-between">
          <Button
              title={t('cancel')}
              variant="critical"
              outline
              onAction={() => {
                window.location = gridUrl;
              }}
          />
          <Button
              title={t('save')}
              onAction={() => {
                document
                    .getElementById('couponForm')
                    .dispatchEvent(
                        new Event('submit', { cancelable: true, bubbles: true })
                    );
              }}
              isLoading={state === 'submitting'}
          />
        </div>
      </>
  );
}

FormContent.propTypes = {
  gridUrl: PropTypes.string.isRequired
};

export const layout = {
  areaId: 'couponForm',
  sortOrder: 10
};

export const query = `
  query Query {
    gridUrl: url(routeId: "couponGrid")
  }
`;
