// @evershop/evershop/src/modules/promotion/pages/admin/couponEdit+couponNew/OrderCondition.jsx
import PropTypes from 'prop-types';
import React from 'react';
import { Input } from '@components/common/form/fields/Input';
import { RequiredProducts } from '@components/admin/promotion/couponEdit/RequireProducts';
import Layout from "@evershop/evershop/src/modules/cms/pages/frontStore/all/Layout";
import {useTranslation} from "react-i18next";
import '../../../../i18n'

export default function OrderCondition({ coupon = {} }) {
    const condition = coupon?.condition || {};
    const {t, i18n} = useTranslation()
    return (
        <div>
            <Input
                name="condition[order_total]"
                label= {t('minumumpurschaseamount')}
                placeholder= {t('enterminumumpurchase')}
                value={condition.orderTotal || null}
            />
            <Input
                name="condition[order_qty]"
                label={t('minpurchaseqty')}
                placeholder={t('enterminqty')}
                value={condition.orderQty || null}
            />
            <RequiredProducts requiredProducts={condition.requiredProducts || []} />
        </div>
    );
}

OrderCondition.propTypes = {
    coupon: PropTypes.shape({
        condition: PropTypes.shape({
            orderTotal: PropTypes.number,
            orderQty: PropTypes.number,
            requiredProducts: PropTypes.arrayOf(
                PropTypes.shape({
                    key: PropTypes.string,
                    operator: PropTypes.string,
                    value: PropTypes.string,
                    qty: PropTypes.string
                })
            )
        })
    })
};

OrderCondition.defaultProps = {
    coupon: {}
};

export const layout = {
    areaId: 'couponEditLeft',
    sortOrder: 10
};

export const query = `
  query Query {
    coupon(id: getContextValue('couponId', null)) {
      condition {
        orderTotal
        orderQty
        requiredProducts {
          key
          operator
          value
          qty
        }
      }
    }
  }
`;
