// @evershop/evershop/src/modules/catalog/pages/admin/attributeEdit+attributeNew/Avaibility.jsx
import PropTypes from 'prop-types';
import React from 'react';
import { Field } from '@components/common/form/Field';
import { Card } from '@components/admin/cms/Card';
import Layout from "@evershop/evershop/src/modules/cms/pages/frontStore/all/Layout";
import {useTranslation} from "react-i18next";
import '../../../../i18n'


export default function General({ attribute }) {
    const {t, i18n} = useTranslation()
  return (
    <Card title= {t('settingname')} subdued>
      <Card.Session>
        <Field
          id="is_required"
          type="radio"
          name="is_required"
          label= {t('isrequired')}
          options={[
            { value: 0, text: t('notrequired') },
            { value: 1, text: t('required') }
          ]}
          value={attribute?.isRequired}
        />
      </Card.Session>
      <Card.Session>
        <Field
          id="is_filterable"
          type="radio"
          name="is_filterable"
          label= {t('isfilterable')}
          options={[
            { value: 0, text:  t('no') },
            { value: 1, text: t('yes') }
          ]}
          value={attribute?.isFilterable}
        />
      </Card.Session>
      <Card.Session>
        <Field
          id="display_on_frontend"
          type="radio"
          name="display_on_frontend"
          label= {t('showtocustomers')}
          options={[
            { value: 0, text: t('no') },
            { value: 1, text: t('yes') }
          ]}
          value={attribute?.displayOnFrontend}
        />
      </Card.Session>
      <Card.Session>
        <Field
          id="sort_order"
          type="text"
          name="sort_order"
          label= {t('sortorder')}
          value={attribute?.sortOrder}
          validationRules={['notEmpty', 'number']}
        />
      </Card.Session>
    </Card>
  );
}

General.propTypes = {
  attribute: PropTypes.shape({
    displayOnFrontend: PropTypes.number,
    isFilterable: PropTypes.number,
    isRequired: PropTypes.number,
    sortOrder: PropTypes.number
  })
};

General.defaultProps = {
  attribute: {}
};

export const layout = {
  areaId: 'rightSide',
  sortOrder: 10
};

export const query = `
  query Query {
    attribute(id: getContextValue("attributeId", null)) {
      attributeId
      isFilterable
      isRequired
      displayOnFrontend
      sortOrder
    }
  }
`;
