// @evershop/evershop/src/modules/catalog/pages/admin/attributeEdit+attributeNew/General.jsx
/* eslint-disable camelcase */
/* eslint-disable jsx-a11y/anchor-is-valid */
import PropTypes from 'prop-types';
import React from 'react';
import Select from 'react-select';
import { useQuery } from 'urql';
import Area from '@components/common/Area';
import { get } from '@evershop/evershop/src/lib/util/get';
import { Field } from '@components/common/form/Field';
import { Card } from '@components/admin/cms/Card';
import { Input } from '@components/common/form/fields/Input';
import Layout from "@evershop/evershop/src/modules/cms/pages/frontStore/all/Layout";
import {useTranslation} from "react-i18next";
import '../../../../i18n'

const GroupsQuery = `
  query Query {
    attributeGroups {
      items {
        value: attributeGroupId
        label: groupName
      }
    }
  }
`;
function Groups({ groups, createGroupApi }) {
  const {t, i18n} = useTranslation()
  const [result, reexecuteQuery] = useQuery({
    query: GroupsQuery
  });
  const newGroup = React.useRef(null);
  const [createGroupError, setCreateGroupError] = React.useState(null);
  const { data, fetching, error } = result;

  const createGroup = () => {
    if (!newGroup.current.value) {
      setCreateGroupError('Group name is required');
      return;
    }
    fetch(createGroupApi, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({ group_name: newGroup.current.value })
    })
      .then((response) => response.json())
      .then((jsonData) => {
        if (!jsonData.error) {
          newGroup.current.value = '';
          reexecuteQuery({ requestPolicy: 'network-only' });
        } else {
          setCreateGroupError(jsonData.error.message);
        }
      });
  };

  if (fetching) return <p>t('loading')...</p>;
  if (error) {
    return (
      <p>
        Oh no...
        {error.message}
      </p>
    );
  }

  return (
    <div>
      <div className="mb-1"> {t('selectattribute')}</div>
      <div className="grid gap-2 grid-cols-2">
        <div>
          <Select
            name="groups[]"
            options={data.attributeGroups.items}
            hideSelectedOptions
            isMulti
            defaultValue={groups}
          />
        </div>
        <div className="grid gap-2 grid-cols-1">
          <div>
            <Input
              type="text"
              placeholder= {t('createnewgroup')}
              ref={newGroup}
              error={createGroupError}
              suffix={
                <a
                  className="text-interactive"
                  href="#"
                  onClick={(e) => {
                    e.preventDefault();
                    createGroup();
                  }}
                >
                  <svg
                    width="1.5rem"
                    height="1.5rem"
                    xmlns="http://www.w3.org/2000/svg"
                    fill="none"
                    viewBox="0 0 24 24"
                    strokeWidth={1.5}
                    stroke="currentColor"
                    className="w-6 h-6"
                  >
                    <path
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      d="M12 9v6m3-3H9m12 0a9 9 0 11-18 0 9 9 0 0118 0z"
                    />
                  </svg>
                </a>
              }
            />
          </div>
        </div>
      </div>
    </div>
  );
}

Groups.propTypes = {
  createGroupApi: PropTypes.string.isRequired,
  groups: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.number,
      label: PropTypes.string
    })
  ).isRequired
};

function Options({ originOptions = [] }) {
  const [options, setOptions] = React.useState(originOptions);

  const addOption = (e) => {
    e.preventDefault();
    setOptions(
      options.concat({
        optionId: Math.floor(Math.random() * (9000000 - 1000000)) + 1000000,
        uuid: Math.floor(Math.random() * (9000000 - 1000000)) + 1000000,
        optionText: ''
      })
    );
  };

  const removeOption = (uuid, e) => {
    e.preventDefault();
    const newOptions = options.filter((option) => option.uuid !== uuid);
    setOptions(newOptions);
  };

  return (
    <div className="attribute-edit-options">
      {options.map((option, index) => {
        const { uuid, optionId, optionText } = option;
        return (
          <div key={uuid} className="flex mb-05 space-x-2">
            <div>
              <Field
                key={uuid}
                type="text"
                name={`options[${index}][option_text]`}
                formId="attribute-edit-form"
                value={optionText}
                validationRules={['notEmpty']}
              />
              <input
                type="hidden"
                name={`options[${index}][option_id]`}
                value={optionId}
              />
            </div>
            <div className="self-center">
              <a
                href="#"
                onClick={(e) => removeOption(uuid, e)}
                className="text-critical hover:underline"
              >
                {t('removeoption')}
              </a>
            </div>
          </div>
        );
      })}
      <div className="mt-1">
        <a
          href="#"
          onClick={(e) => addOption(e)}
          className="text-interactive hover:underline"
        >
          {t('addoption')}
        </a>
      </div>
    </div>
  );
}

Options.propTypes = {
  originOptions: PropTypes.arrayOf(
    PropTypes.shape({
      optionId: PropTypes.number,
      optionText: PropTypes.string
    })
  ).isRequired
};

export default function General({ attribute, createGroupApi }) {
  const {t, i18n} = useTranslation()
  const [type, setType] = React.useState(attribute?.type);
  const fields = [
    {
      component: { default: Field },
      props: {
        id: 'attributeName',
        name: 'attribute_name',
        label: t('name'),
        validationRules: ['notEmpty'],
        type: 'text'
      },
      sortOrder: 10
    },
    {
      component: { default: Field },
      props: {
        id: 'attributeCode',
        name: 'attribute_code',
        label: t('attributecode'),
        validationRules: ['notEmpty'],
        type: 'text'
      },
      sortOrder: 15
    },
    {
      component: { default: Field },
      props: {
        id: 'attributeId',
        name: 'attribute_id',
        type: 'hidden'
      },
      sortOrder: 10
    },
    {
      component: { default: Field },
      props: {
        id: 'type',
        type: 'radio',
        name: 'type',
        label: t('type'),
        options: [
          { value: 'text', text: t('text') },
          { value: 'select', text: t('select') },
          { value: 'multiselect', text: t('multiselect') },
          { value: 'textarea', text: t('textarea') }
        ],
        onChange: (value) => {
          setType(value);
        }
      },
      sortOrder: 20
    }
  ].map((f) => {
    if (get(attribute, `${f.props.id}`) !== undefined) {
      // eslint-disable-next-line no-param-reassign
      f.props.value = get(attribute, `${f.props.id}`);
    }
    return f;
  });

  return (
    <Card title= {t('general')}>
      <Card.Session>
        <Area id="attributeEditGeneral" coreComponents={fields} />
      </Card.Session>
      {['select', 'multiselect'].includes(type) && (
        <Card.Session title="Attribute options">
          <Options originOptions={get(attribute, 'options', [])} />
        </Card.Session>
      )}
      <Card.Session title= {t('attributegroup')}>
        <Groups
          groups={get(attribute, 'groups', [])}
          createGroupApi={createGroupApi}
        />
      </Card.Session>
    </Card>
  );
}

General.propTypes = {
  attribute: PropTypes.shape({
    type: PropTypes.string.isRequired,
    attribute_id: PropTypes.number,
    attribute_name: PropTypes.string,
    attribute_code: PropTypes.string,
    options: PropTypes.arrayOf(
      PropTypes.shape({
        optionId: PropTypes.number,
        optionText: PropTypes.string
      })
    ),
    groups: PropTypes.arrayOf(
      PropTypes.shape({
        value: PropTypes.number,
        label: PropTypes.string
      })
    )
  }),
  createGroupApi: PropTypes.string.isRequired
};

General.defaultProps = {
  attribute: {
    type: 'text'
  }
};

export const layout = {
  areaId: 'leftSide',
  sortOrder: 10
};

export const query = `
  query Query {
    attribute(id: getContextValue("attributeId", null)) {
      attributeId
      attributeName
      attributeCode
      type
      options {
        optionId: attributeOptionId
        uuid
        optionText
      }
      groups {
        value: attributeGroupId
        label: groupName
      }
    }
    createGroupApi: url(routeId: "createAttributeGroup")
  }
`;
