// @evershop/evershop/src/modules/catalog/pages/admin/attributeEdit+attributeNew/PageHeading.jsx
import PropTypes from 'prop-types';
import React from 'react';
import PageHeading from '@components/admin/cms/PageHeading';
import Layout from "@evershop/evershop/src/modules/cms/pages/frontStore/all/Layout";
import {useTranslation} from "react-i18next";
import '../../../../i18n'

export default function AttributeEditPageHeading({ backUrl, attribute }) {
  const {t, i18n} = useTranslation()
  return (
    <PageHeading
      backUrl={backUrl}
      heading={
        attribute
          ? `Editing ${attribute.attributeName}`
          : t('createnewattribute')
      }
    />
  );
}

AttributeEditPageHeading.propTypes = {
  attribute: PropTypes.shape({
    attributeName: PropTypes.string
  }),
  backUrl: PropTypes.string.isRequired
};

AttributeEditPageHeading.defaultProps = {
  attribute: {}
};

export const layout = {
  areaId: 'content',
  sortOrder: 5
};

export const query = `
  query Query {
    attribute(id: getContextValue("attributeId", null)) {
      attributeName
    }
    backUrl: url(routeId: "attributeGrid")
  }
`;
