// @evershop/evershop/src/modules/catalog/pages/admin/collectionEdit+collectionNew/FormContent.jsx
import PropTypes from 'prop-types';
import React from 'react';
import Area from '@components/common/Area';
import Button from '@components/common/form/Button';
import { useFormContext } from '@components/common/form/Form';
import './FormContent.scss';
import Layout from "@evershop/evershop/src/modules/cms/pages/frontStore/all/Layout";
import {useTranslation} from "react-i18next";
import '../../../../i18n'

export default function FormContent({ gridUrl }) {
    const {t, i18n} = useTranslation()
  const { state } = useFormContext();
  return (
    <div className="w-2/3" style={{ margin: '0 auto' }}>
      <div className="grid gap-2">
        <Area id="collectionFormInner" noOuter />
      </div>
      <div className="form-submit-button flex border-t border-divider mt-15 pt-15 justify-between">
        <Button
          title= {t('cancel')}
          variant="critical"
          outline
          onAction={() => {
            window.location = gridUrl;
          }}
        />
        <Button
          title= {t('save')}
          onAction={() => {
            document
              .getElementById('collectionForm')
              .dispatchEvent(
                new Event('submit', { cancelable: true, bubbles: true })
              );
          }}
          isLoading={state === 'submitting'}
        />
      </div>
    </div>
  );
}

FormContent.propTypes = {
  gridUrl: PropTypes.string.isRequired
};

export const layout = {
  areaId: 'collectionForm',
  sortOrder: 10
};

export const query = `
  query Query {
    gridUrl: url(routeId: "collectionGrid")
  }
`;
