// @evershop/evershop/src/modules/catalog/pages/admin/collectionEdit+collectionNew/PageHeading.jsx
import PropTypes from 'prop-types';
import React from 'react';
import PageHeading from '@components/admin/cms/PageHeading';
import Layout from "@evershop/evershop/src/modules/cms/pages/frontStore/all/Layout";
import {useTranslation} from "react-i18next";
import '../../../../i18n'

export default function CollectionEditPageHeading({ backUrl, collection }) {
  const {t, i18n} = useTranslation()
  return (
    <div className='w-2/3' style={{margin: '0 auto'}}>
      <PageHeading
        backUrl={backUrl}
        heading={collection ? `Editing ${collection.name}` : t('createnewcollection')}
      />
    </div>
  );
}

CollectionEditPageHeading.propTypes = {
  backUrl: PropTypes.string.isRequired,
  collection: PropTypes.shape({
    name: PropTypes.string
  })
};

CollectionEditPageHeading.defaultProps = {
  collection: {}
};

export const layout = {
  areaId: 'content',
  sortOrder: 5
};

export const query = `
  query Query {
    collection(code: getContextValue("collectionCode", null)) {
      name
    }
    backUrl: url(routeId: "collectionGrid")
  }
`;
