// @evershop/evershop/src/modules/catalog/pages/admin/productEdit+productNew/Inventory.jsx
import PropTypes from 'prop-types';
import React from 'react';
import { Field } from '@components/common/form/Field';
import { Card } from '@components/admin/cms/Card';
import Layout from "@evershop/evershop/src/modules/cms/pages/frontStore/all/Layout";
import {useTranslation} from "react-i18next";
import '../../../../i18n'


export default function Inventory({ product }) {
    const {t, i18n} = useTranslation()
  const inventory = product?.inventory || {};
  return (
    <Card title= {t('inventory')} subdued>
      <Card.Session>
        <Field
          id="manage_stock"
          name="manage_stock"
          value={inventory.manageStock}
          label= {t('managestock')}
          options={[
            { value: 0, text: t('no') },
            { value: 1, text: t('yes') }
          ]}
          type="radio"
        />
      </Card.Session>
      <Card.Session>
        <Field
          id="stock_availability"
          name="stock_availability"
          value={inventory.stockAvailability}
          label= {t('stockavailability')}
          options={[
            { value: 0, text: t('no') },
            { value: 1, text: t('yes') }
          ]}
          type="radio"
        />
      </Card.Session>
      <Card.Session>
        <Field
          id="qty"
          name="qty"
          value={inventory.qty}
          placeholder={t('quantity')}
          label= {t('quantity')}
          type="text"
          validationRules={['notEmpty']}
        />
      </Card.Session>
    </Card>
  );
}

Inventory.propTypes = {
  product: PropTypes.shape({
    inventory: PropTypes.shape({
      qty: PropTypes.number,
      stockAvailability: PropTypes.number,
      manageStock: PropTypes.number
    })
  })
};

Inventory.defaultProps = {
  product: {
    inventory: {
      qty: 0,
      stockAvailability: 0,
      manageStock: 0
    }
  }
};

export const layout = {
  areaId: 'rightSide',
  sortOrder: 15
};

export const query = `
  query Query {
    product(id: getContextValue("productId", null)) {
      inventory {
        qty
        stockAvailability
        manageStock
      }
    }
  }
`;
