// @evershop/evershop/src/modules/catalog/pages/admin/productEdit+productNew/Seo.jsx
import PropTypes from 'prop-types';
import React from 'react';
import Area from '@components/common/Area';
import { Field } from '@components/common/form/Field';
import { get } from '@evershop/evershop/src/lib/util/get';
import { Card } from '@components/admin/cms/Card';
import Layout from "@evershop/evershop/src/modules/cms/pages/frontStore/all/Layout";
import {useTranslation} from "react-i18next";
import '../../../../i18n'

export default function SEO({ product }) {
  const {t, i18n} = useTranslation()
  const fields = [
    {
      component: { default: Field },
      props: {
        id: 'urlKey',
        name: 'url_key',
        label: t('urlkey'),
        validationRules: ['notEmpty'],
        type: 'text'
      },
      sortOrder: 0
    },
    {
      component: { default: Field },
      props: {
        id: 'metaTitle',
        name: 'meta_title',
        label: t('metatitle'),
        type: 'text'
      },
      sortOrder: 10
    },
    {
      component: { default: Field },
      props: {
        id: 'metaKeywords',
        name: 'meta_keywords',
        label: t('metakeywords'),
        type: 'text'
      },
      sortOrder: 20
    },
    {
      component: { default: Field },
      props: {
        id: 'metaDescription',
        name: 'meta_description',
        label: t('metadescription'),
        options: [
          { value: 0, text: t('disabled') },
          { value: 1, text: t('enabled')  }
        ],
        type: 'textarea'
      },
      sortOrder: 30
    }
  ].map((f) => {
    if (get(product, `${f.props.id}`) !== undefined) {
      // eslint-disable-next-line no-param-reassign
      f.props.value = get(product, `${f.props.id}`);
    }
    return f;
  });

  return (
    <Card title= {t('searchengine')}>
      <Card.Session>
        <Area id="productEditSeo" coreComponents={fields} />
      </Card.Session>
    </Card>
  );
}

SEO.propTypes = {
  product: PropTypes.shape({
    urlKey: PropTypes.string,
    metaTitle: PropTypes.string,
    metaKeywords: PropTypes.string,
    metaDescription: PropTypes.string
  })
};

SEO.defaultProps = {
  product: {
    urlKey: '',
    metaTitle: '',
    metaKeywords: '',
    metaDescription: ''
  }
};

export const layout = {
  areaId: 'leftSide',
  sortOrder: 60
};

export const query = `
  query Query {
    product(id: getContextValue('productId', null)) {
      urlKey
      metaTitle
      metaKeywords
      metaDescription
    }
  }
`;
