// @evershop/evershop/src/modules/catalog/pages/admin/productEdit+productNew/Status.jsx
import PropTypes from 'prop-types';
import React from 'react';
import { Field } from '@components/common/form/Field';
import { Card } from '@components/admin/cms/Card';
import Layout from "@evershop/evershop/src/modules/cms/pages/frontStore/all/Layout";
import {useTranslation} from "react-i18next";
import '../../../../i18n'

export default function Status({ product }) {
    const {t, i18n} = useTranslation()
  return (
    <Card title= {t('productstatus')} subdued>
      <Card.Session>
        <Field
          id="status"
          name="status"
          value={product?.status}
          label= {t('Status')}
          options={[
            { value: 0, text: t('disabled') },
            { value: 1, text: t('enabled') }
          ]}
          type="radio"
        />
      </Card.Session>
      <Card.Session>
        <Field
          id="visibility"
          name="visibility"
          value={product?.visibility}
          label= {t('visibility')}
          options={[
            { value: 0, text: t('notvisibility') },
            { value: 1, text: t('visible') }
          ]}
          type="radio"
        />
      </Card.Session>
    </Card>
  );
}

Status.propTypes = {
  product: PropTypes.shape({
    status: PropTypes.number.isRequired,
    visibility: PropTypes.number.isRequired
  })
};

Status.defaultProps = {
  product: {
    status: 1,
    visibility: 1
  }
};

export const layout = {
  areaId: 'rightSide',
  sortOrder: 10
};

export const query = `
  query Query {
    product(id: getContextValue("productId", null)) {
      status
      visibility
      category {
        value: categoryId
        label: name
      }
    }
  }
`;
