// @evershop/evershop/src/modules/i18n.js
import i18n from "i18next";
import { initReactI18next } from "react-i18next";



const resources = {
    tr: {
        translation: {
            welcome: 'hoşgeldin',
            fullName: 'isim',
            Email: 'E-Posta',
            Status: 'Durum',
            CreatedAt :'Oluşturuldu',
            textwrite: 'Gösterilecek müşteri yok',
            heading: 'Müşteriler',
            Perpage: 'Sayfa Başı',
            show: 'Görüntüle',
            records: 'Kayıtlar',
            searchinput: 'Arama',
            couponcode: 'Kupon Kodu',
            startdate: 'Başlama Tarihi',
            enddate: 'Bitiş Tarihi',
            usedtimes: 'Kullanılan Zamanlar',
            textwritecoupon: 'Görüntülenecek Kupon Yok',
            headingcoupon: 'Kuponlar',
            newcoupon: 'Yeni Kupon',
            settingname: 'Ayarlar',
            name: 'İsim',
            textwritepage: 'Görüntülenecek sayfa yok',
            cmspage: 'Cms Sayfaları',
            newpage: 'Yeni Sayfa',
            general: 'Genel',
            discounttype: 'İndirim Türü',
            ordercontions: 'Sipariş Koşulları',
            customercontions: 'Müşteri Koşulları',
            cancel: 'İptal Etmek',
            save: 'Kaydetmek',
            entercouponcode: 'Kupon kodunu girin',
            description: 'Açıklama',
            enterdescription: 'Açıklamayı girin',
            freeshipiing: 'Ücretsiz Kargo?',
            customeremailempty: 'Müşteri e-postası (hepsi için boş)',
            entercustomer: 'Müşteri e-postalarını girin',
            usecommaemail: 'Çoklu e-postanız varsa virgül kullanın',
            customerspurchase: "Müşterinin satın alması",
            enterpurschased: 'Satın alınan tutarı girin',
            minumumtext: 'Minimum satın alma tutarı',
            fixeddiscount: 'Tüm siparişte sabit indirim',
            percentagediscount: 'Tüm siparişte yüzde indirim',
            fixeddiscountproduct: 'Belirli ürünlerde sabit indirim',
            percentageproduct: 'Belirli ürünlere yüzde indirim',
            buyxgety: 'X satın al, Y kazan',
            minumumpurschaseamount: "Minimum satın alma tutarı",
            enterminumumpurchase: "Minimum satın alma tutarını girin",
            minpurchaseqty: "Minimum satın alma miktarı",
            enterminqty: "Minimum satın alma miktarını girin",
            createnewcoupon: 'Yeni Bir Kupon Oluşturun',
            paymentsetting: 'Ödeme Ayarı',
            configureavaible: 'Mevcut ödeme yöntemlerini yapılandırın',
            storesetting: 'Mağaza Ayarı',
            storeinformation: 'Mağaza bilgilerinizi yapılandırın',
            taxsetting:'Vergi Ayarı',
            taxrates: 'Vergi sınıflarını ve vergi oranlarını yapılandırın',
            province: 'Vilayet',
            postalcode: "Posta Kodu",
            city: 'Şehir',
            address: 'Adres',
            storeadres: 'Mağaza Adresi',
            storeemail: "E-posta'yı sakla",
            storePhoneNumber: 'Telefon Numarasını Sakla',
            contacinformation: 'İLETİŞİM BİLGİLERİ',
            storedescription: "Mağaza Açıklaması",
            storename: 'Mağaza Adı',
            storeinformationsetting: 'BİLGİ DEPOLA',
            settingsaved: 'Ayar Kaydedildi',
            country: 'Ülke',
            salestatic: "Satış İstatistikleri",
            daily: 'Günlük',
            weekly: 'Haftalık',
            monthly: 'Aylık',
            lifetimesales:  "Ömür Boyu Satış",
            bestsellers: "En çok satanlar",
            allproduct: 'Tüm ürünler',
            statictext: "Yeni başlamış gibisin. Henüz en çok satanlar yok.",
            bestcustomer: "En iyi müşteriler",
            allcustomers: 'Tüm müşteriler',
            orders: 'Siparişler',
            total: 'Toplam',
            comploted: 'tamamlanan siparişlerin sayısı',
            cancelled: 'iptal edilen siparişlerin sayısı',
            shipping: 'NAKLİYE AYARI',
            shippingtext: 'Gönderim yaptığınız yer, gönderim yöntemleri ve teslimat ücreti',
            sale: 'Satış',
            productname: "Ürün Adı",
            price: 'Fiyat',
            sku: 'Stok Kodu',
            qty: 'Adet',
            producttextdown: 'Gösterilecek ürün yok',
            products: 'Ürünler',
            newproduct: 'Yeni Ürün',
            weight: 'Ağırlık',
            category: 'Kategori',
            selectcategory: 'Kategori Seç',
            change: 'Değiştirmek',
            Unassign: 'Atamayı kaldır',
            attribute: "Özellik grubu",
            attributestext: 'Zaten mevcut olan bir ürünün özellik grubu değiştirilemez bir varyant grubunda.',
            attributes: 'Nitelikler',
            managestock: "Stok yönetilsin mi?",
            inventory: "Envanter",
            yes: 'Evet',
            no: 'Hayır',
            stockavailability: "Stokta bulunma durumu",
            quantity: "Miktar",
            media: 'Medya',
            createnewproduct: 'Yeni Bir Ürün Oluştur',
            urlkey: 'Url Anahtarı',
            metatitle: 'Meta Başlığı',
            metakeywords: 'Anahtar Kelimeler',
            metadescription: 'Meta Açıklaması',
            disabled: 'Engelli',
            enabled: 'Etkinleştirilmiş',
            searchengine: "Arama motoru optimizasyonu",
            productstatus: "Ürün durumu",
            visibility: "Görünürlük",
            notvisibility: "Görünmezlik",
            visible: 'Görünür',
            taxclass: 'Vergi Sınıfı',
            none: 'Hiçbiri',
            categoryname: 'Kategori Adı',
            includeinmenu: "Menüye Ekle",
            categorytext: 'Görüntülenecek Kategori Yok',
            categories: 'Kategoriler',
            newcategory: 'Yeni Kategori',
            parentcategory: 'Aile kategorisi',
            categorybanner: "Kategori Banner'ı",
            remove: 'Kaldırmak',
            addimage: 'Resim Eklemek',
            clickuploadimage: 'resim yüklemek için tıklayın',
            createcategory: 'Yeni Bir Kategori Oluştur',
            includeinstoremenu: "Mağaza Menüsüne Ekle",
            id: "ID",
            collectionname: "Koleksiyon Adı",
            code: 'Kod',
            collectiontext: 'Görüntülenecek koleksiyon yok',
            collections: 'Koleksiyonlar',
            newcollection: "Yeni Koleksiyon" ,
            addproducts: 'Ürün Ekle',
            searchproducts: "Ürünleri ara",
             productnotext: "Gösterilecek ürün yok.",
            previous: "Önceki",
            next: "Sonraki",
            items: "öğeler",
            featuredproducts: 'Özel Ürünler',
            uniqueid: 'Benzersiz Kimlik',
            createnewcollection: 'Yeni Koleksiyon Oluştur',
            delete: 'Sil',
            loading: 'yükleniyor',
            attributename: "Özellik Adı",
            type: 'Tür',
            isrequired: "Gerekli mi?",
            isfilterable: "Filtrelenebilir mi?",
            select: 'Seç',
            attributetext: "Görüntülenecek özellik yok",
            newattribute: 'Yeni Özellik',
            required: 'Gerekli',
            notrequired: 'Gerekli değil',
            showtocustomers:"Müşterilere gösterilsin mi?",
            sortorder: "Sıralama düzeni",
            createnewgroup: "Yeni bir grup oluştur",
            selectattribute: 'Özelliğin ait olduğu grupları seç',
            removeoption: 'Seçeneği kaldır',
            addoption: 'Seçenek ekle',
            attributecode: 'Özellik kodu',
            text: 'Metin',
            multiselect: 'Çoklu seçim',
            textarea: 'Metin alanı',
            attributegroup: "Özellik Grubu",
            createnewattribute: 'Yeni Bir Özellik Oluştur',
            ordernumber: "Sipariş Numarası",
            date: 'Tarih',
            customeremail:"Müşteri E-postası",
            shipmentstatus: "Gönderi Durumu",
            paymentstatus: "Ödeme Durumu",
            ordertext: "Görüntülenecek bir sipariş yok",
            promotion: 'Terfi',
            customer: 'Müşteri',
            pages: 'Sayfalar',
            catalog: 'Katalog',
            dashboard: 'Kontrol Paneli',
            quicklinks: "Hızlı Linkler",
            searchtextv: 'Ürünleri, siparişi ve diğer kaynakları arayın'















        }
    },
    en: {
        translation: {
            welcome: 'welcome',
            fullName: 'fullname',
            Email: 'Email',
            Status: 'Status',
            CreatedAt :'Created At',
            textwrite: 'There is no customer to display',
            heading: 'Customers',
            Perpage: 'Per Page',
            show: 'Show',
            records: 'Records',
            searchinput: 'Search',
            couponcode: 'Coupon Code',
            startdate: 'Start Date',
            enddate: 'End Date',
            usedtimes: 'Used Times',
            textwritecoupon: 'There is no coupon to display',
            headingcoupon: 'Coupons',
            newcoupon: 'New Coupon',
            settingname: 'Setting',
            name: 'Name',
            textwritepage: 'There is no coupon to display',
            cmspage: 'Cms Pages',
            newpage: 'New Page',
            general: 'General',
            discounttype: 'Discount Type',
            ordercontions: 'Order conditions',
            customercontions: 'Customer conditions',
            cancel: 'Cancel',
            save: 'Save',
            entercouponcode: 'Enter coupon code',
            description: 'Description',
            enterdescription: 'Enter description',
            freeshipiing: 'Free shipping?',
            customeremailempty: 'Customer email (empty for all)',
            entercustomer: 'Enter customer emails',
            usecommaemail: 'Use comma when you have multi email',
            customerspurchase: "Customer's purchase",
            enterpurschased: 'Enter purchased amount',
            minumumtext: 'Minimum purchased amount',
            fixeddiscount: 'Fixed discount to entire order',
            percentagediscount: 'Percentage discount to entire order',
            fixeddiscountproduct: 'Fixed discount to specific products',
            percentageproduct: 'Percentage discount to specific products',
            buyxgety: 'Buy X get Y',
            minumumpurschaseamount: "Minimum purchase amount",
            enterminumumpurchase: "Enter minimum purchase amount",
            minpurchaseqty: "Minimum purchase qty",
            enterminqty: "Enter minimum purchase qty",
            createnewcoupon: 'Create A New Coupon',
            paymentsetting: 'Payment Setting',
            configureavaible: 'Configure the available payment methods',
            storesetting: 'Store Setting',
            storeinformation: 'Configure your store information',
            taxsetting: 'Tax Setting',
            taxrates: 'Configure tax classes and tax rates',
            province: 'Province',
            postalcode: "PostalCode",
            city: 'City',
            address: 'Address',
            storeadres: 'Store Address',
            storeemail: "Store Email",
            storePhoneNumber: 'Store Phone Number',
            contacinformation: 'CONTACT INFORMATION',
            storedescription: 'Store Description',
            storename: 'Store Name',
            storeinformationsetting: 'STORE INFORMATİON',
            settingsaved: 'Setting Saved',
            country: 'Country',
            salestatic: "Sale Statistics",
            daily: 'Daily',
            weekly: 'Weekly',
            monthly: 'Monthly',
           lifetimesales:  "Lifetime Sales",
            bestsellers: "Best Sellers",
            allproduct: 'All products',
            statictext: "Look like you just started. No bestsellers yet.",
            bestcustomer: "Best customers",
            allcustomers: 'All customers',
            orders: 'Orders',
            total: 'Total',
            comploted: 'of orders completed',
            cancelled: 'of orders cancelled',
            shipping: 'SHIPPING SETTING',
            shippingtext: 'Where you ship, shipping methods and delivery fee',
            sale: 'Sale',
            productname: "Product Name",
            price: 'Price',
            sku: 'SKU',
            qty: 'Qty',
            producttextdown: 'There is no product to display',
            products: 'Products',
            newproduct: 'New Product',
            weight: 'Weight',
            category: 'Category',
            selectcategory: 'Select Category',
            change: 'Change',
            Unassign: 'Unassign',
            attribute: "Attribute group",
            attributestext: 'Can not change the attribute group of a product that is already in a variant group.',
            attributes: 'Attributes',
            managestock: "Manage stock?",
            inventory: "Inventory",
            yes: 'Yes',
            no: 'No',
            stockavailability: "Stock availability",
            quantity: "Quantity",
            media: 'Media',
            createnewproduct: 'Create A New Product',
            urlkey: 'Url Key',
            metatitle: 'Meta Title',
            metakeywords: 'Meta Keywords',
            metadescription: 'Meta Description',
            disabled: 'Disabled',
            enabled: 'Enabled',
            searchengine: "Search engine optimize",
            productstatus: "Product status",
            visibility: "Visibility",
            notvisibility: "Not Visibility",
            visible: 'Visible',
            taxclass: 'Tax Class',
            none: 'None',
            categoryname: 'Category Name',
            includeinmenu: "Include In Menu",
            categorytext: 'There is no category to display',
            categories: 'Categories',
            newcategory: 'New Category',
            parentcategory: 'Parent category',
            categorybanner: "Category Banner",
            remove: 'Remove',
            addimage: "Add image",
            clickuploadimage: 'click to upload an image',
            createcategory: 'Create A New category',
            includeinstoremenu: "Include In Store Menu",
            id: "ID",
            collectionname: "Collection Name",
            code: 'Code',
             collectiontext: 'There is no collections to display',
            collections: 'Collections',
            newcollection: "New Collection",
            addproducts: 'Add products',
            searchproducts: "Search products",
            productnotext: "No product to display.",
            previous: "Previous",
            next: "Next",
            items: "items",
            featuredproducts:'Featured Products',
            uniqueid: 'Unique ID',
            createnewcollection: 'Create A New Collection',
            delete: 'Delete',
            loading: 'isLoading',
            attributename: "Attribute Name",
            type: 'Type',
            isrequired: "Is Required?",
            isfilterable: "Is Filterable?",
            select: 'Select',
           attributetext: "There is no attribute to display",
            newattribute: 'New Attribute',
            required: 'Required',
           notrequired: 'Not required',
            showtocustomers: "Show to customers?",
            sortorder: "Sort order",
            createnewgroup: "Create a new group",
            selectattribute: 'Select groups the attribute belongs to',
            removeoption: ' Remove option',
            addoption: 'Add Option',
            attributecode: 'Attribute code',
            text: 'Text',
            multiselect: 'Multiselect',
            textarea: 'Textarea',
            attributegroup: "Attribute Group",
            createnewattribute: 'Create A New Attribute',
            ordernumber: "Order Number",
            date: 'Date',
            customeremail:"Customer Email",
           shipmentstatus: "Shipment Status",
            paymentstatus: "Payment Status",
            ordertext: "There is no order to display",
            promotion: 'Promotion',
            customer: 'Customer',
            pages: 'Pages',
            catalog: 'Catalog',
            dashboard: 'Dashboard',
            quicklinks: "Quick links",
            searchtextv: 'Search for products, order and other resources'


        }
    }
};

i18n
    .use(initReactI18next)
    .init({
        lng: 'en',
        resources
    });

export default i18n;
