// @evershop/evershop/src/modules/catalog/pages/admin/categoryEdit+categoryNew/Status.jsx
import PropTypes from 'prop-types';
import React from 'react';
import { Field } from '@components/common/form/Field';
import { Card } from '@components/admin/cms/Card';
import Layout from "@evershop/evershop/src/modules/cms/pages/frontStore/all/Layout";
import {useTranslation} from "react-i18next";
import '../../../../i18n'

export default function Status({ category }) {
    const {t, i18n} = useTranslation()
  return (
    <Card>
      <Card.Session>
        <Field
          type="radio"
          name={t('Status')}
          options={[
            { value: 0, text: t('disabled') },
            { value: 1, text: t('enabled') }
          ]}
          value={category?.status}
        />
      </Card.Session>
      <Card.Session title= {t('includeinstoremenu')}>
        <Field
          type="radio"
          name="include_in_nav"
          options={[
            { value: 0, text: t('no') },
            { value: 1, text: t('yes') }
          ]}
          value={category?.includeInNav}
        />
      </Card.Session>
    </Card>
  );
}

Status.propTypes = {
  category: PropTypes.shape({
    status: PropTypes.number,
    includeInNav: PropTypes.number
  })
};

Status.defaultProps = {
  category: {}
};

export const layout = {
  areaId: 'rightSide',
  sortOrder: 15
};

export const query = `
  query Query {
    category(id: getContextValue("categoryId", null)) {
      status
      includeInNav
    }
  }
`;
