// @evershop/evershop/src/modules/setting/pages/admin/storeSetting/StoreSetting.jsx
import React from 'react';
import PropTypes from 'prop-types';
import { useQuery } from 'urql';
import { toast } from 'react-toastify';
import { Field } from '@components/common/form/Field';
import { Form } from '@components/common/form/Form';
import { Card } from '@components/admin/cms/Card';
import SettingMenu from '@components/admin/setting/SettingMenu';
import Layout from "@evershop/evershop/src/modules/cms/pages/frontStore/all/Layout";
import {useTranslation} from "react-i18next";
import '../../../../i18n'


const ProvincesQuery = `
  query Province($countries: [String]) {
    provinces (countries: $countries) {
      code
      name
      countryCode
    }
  }
`;

const CountriesQuery = `
  query Country($countries: [String]) {
    countries (countries: $countries) {
      code
      name
    }
  }
`;

const CurrencyQuery = `
  query Currencies {
    currencies {
      code
      name
    }
  }
`;

function Province({
  selectedCountry = 'US',
  selectedProvince,
  allowedCountries = [],
  fieldName = 'storeProvince'
}) {
  const {t, i18n} = useTranslation()
  const [result] = useQuery({
    query: ProvincesQuery,
    variables: { countries: allowedCountries }
  });
  const { data, fetching, error } = result;

  if (fetching) return <p>Loading...</p>;
  if (error) {
    return (
      <p>
        Oh no...
        {error.message}
      </p>
    );
  }
  const provinces = data.provinces.filter(
    (p) => p.countryCode === selectedCountry
  );
  if (!provinces.length) {
    return null;
  }

  return (
      <div>
        <Field
            type="select"
            value={selectedProvince}
            name={fieldName}
            label= {t('province')}
            placeholder= {t('province')}
            validationRules={['notEmpty']}
            options={provinces.map((p) => ({ value: p.code, text: p.name }))}
        />
      </div>
  );
}

Province.propTypes = {
  allowedCountries: PropTypes.arrayOf(PropTypes.string),
  fieldName: PropTypes.string,
  selectedCountry: PropTypes.string,
  selectedProvince: PropTypes.string
};

Province.defaultProps = {
  allowedCountries: [],
  fieldName: 'storeProvince',
  selectedCountry: 'US',
  selectedProvince: ''
};

function Country({
  selectedCountry,
  setSelectedCountry,
  allowedCountries = [],
  fieldName = 'storeCountry'
}) {
  const {t, i18n} = useTranslation()
  const onChange = (e) => {
    setSelectedCountry(e.target.value);
  };
  const [result] = useQuery({
    query: CountriesQuery,
    variables: { countries: allowedCountries }
  });

  const { data, fetching, error } = result;

  if (fetching) return <p>Loading...</p>;
  if (error) {
    return (
      <p>
        Oh no...
        {error.message}
      </p>
    );
  }

  return (
    <div style={{ marginTop: '1rem' }}>
      <Field
        type="select"
        value={selectedCountry}
        label= {t('country')}
        name={fieldName}
        placeholder={t('country')}
        onChange={onChange}
        validationRules={['notEmpty']}
        options={data.countries.map((c) => ({ value: c.code, text: c.name }))}
      />
    </div>
  );
}

Country.propTypes = {
  allowedCountries: PropTypes.arrayOf(PropTypes.string),
  fieldName: PropTypes.string,
  selectedCountry: PropTypes.string.isRequired,
  setSelectedCountry: PropTypes.func.isRequired
};

Country.defaultProps = {
  allowedCountries: [],
  fieldName: 'storeCountry'
};

function Currency({ selectedCurrency, fieldName = 'storeCurrency' }) {
  const [result] = useQuery({
    query: CurrencyQuery
  });
  const { data, fetching, error } = result;

  if (fetching) return <p>Loading...</p>;
  if (error) {
    return (
      <p>
        Oh no...
        {error.message}
      </p>
    );
  }

  return (
    <Field
      type="select"
      value={selectedCurrency}
      name={fieldName}
      label="Currency"
      placeholder="Currency"
      options={data.currencies.map((c) => ({ value: c.code, text: c.name }))}
    />
  );
}

Currency.propTypes = {
  fieldName: PropTypes.string,
  selectedCurrency: PropTypes.string.isRequired
};

Currency.defaultProps = {
  fieldName: 'storeCurrency'
};

export default function StoreSetting({
  saveSettingApi,
  setting: {
    storeName,
    storeDescription,
    storePhoneNumber,
    storeEmail,
    storeCountry,
    storeAddress,
    storeCity,
    storeProvince,
    storePostalCode
  }
}) {
  const {t, i18n} = useTranslation()
  const [selectedCountry, setSelectedCountry] = React.useState(() => {
    const country = storeCountry;
    if (!country) {
      return 'US';
    } else {
      return country;
    }
  });

  return (
      <div className="main-content-inner">
        <div className="grid grid-cols-6 gap-x-2 grid-flow-row ">
          <div className="col-span-2">
            <SettingMenu />
          </div>
          <div className="col-span-4">
            <Form
                method="POST"
                id="storeSetting"
                action={saveSettingApi}
                onSuccess={(response) => {
                  if (!response.error) {
                    toast.success('Setting saved');
                  } else {
                    toast.error(response.error.message);
                  }
                }}
            >
              <Card>
                <Card.Session title={t('storeinformationsetting')}>
                  <Field
                      name="storeName"
                      label={t('storename')}
                      placeholder={t('storename')}
                      value={storeName}
                      type="text"
                  />
                  <Field
                      name="storeDescription"
                      label={t('storedescription')}
                      placeholder={t('storedescription')}
                      value={storeDescription}
                      type="textarea"
                  />
                </Card.Session>
                <Card.Session title= {t('contacinformation')}>
                  <div className="grid grid-cols-2 gap-2 mt-2">
                    <div>
                      <Field
                          name="storePhoneNumber"
                          label={t('storePhoneNumber')}
                          value={storePhoneNumber}
                          placeholder={t('storePhoneNumber')}
                          type="text"
                      />
                    </div>
                    <div>
                      <Field
                          name="storeEmail"
                          label={t('storeemail')}
                          value={storeEmail}
                          placeholder={t('storeemail')}
                          type="text"
                      />
                    </div>
                  </div>
                </Card.Session>
                <Card.Session title={t('address')}>
                  <Country
                      selectedCountry={storeCountry}
                      setSelectedCountry={setSelectedCountry}
                  />
                  <Field
                      name="storeAddress"
                      label={t('address')}
                      value={storeAddress}
                      placeholder={t('storeadres')}
                      type="text"
                  />
                  <div className="grid grid-cols-3 gap-2 mt-2">
                    <div>
                      <Field
                          name="storeCity"
                          label= {t('city')}
                          value={storeCity}
                          placeholder= {t('city')}
                          type="text"
                      />
                    </div>
                    <Province
                        selectedProvince={storeProvince}
                        selectedCountry={selectedCountry}
                    />
                    <div>
                      <Field
                          name="storePostalCode"
                          label= {t('postalcode')}
                          value={storePostalCode}
                          placeholder={t('postalcode')}
                          type="text"
                      />
                    </div>
                  </div>
                </Card.Session>
              </Card>
            </Form>
          </div>
        </div>
      </div>
  );
}

StoreSetting.propTypes = {
  saveSettingApi: PropTypes.string.isRequired,
  setting: PropTypes.shape({
    storeName: PropTypes.string,
    storeDescription: PropTypes.string,
    storeTimeZone: PropTypes.string,
    storePhoneNumber: PropTypes.string,
    storeEmail: PropTypes.string,
    storeCountry: PropTypes.string,
    storeAddress: PropTypes.string,
    storeCity: PropTypes.string,
    storeProvince: PropTypes.string,
    storePostalCode: PropTypes.string
  }).isRequired
};

export const layout = {
  areaId: 'content',
  sortOrder: 10
};

export const query = `
  query Query {
    saveSettingApi: url(routeId: "saveSetting")
    setting {
      storeName
      storeDescription
      storeTimeZone
      storePhoneNumber
      storeEmail
      storeCountry
      storeAddress
      storeCity
      storeProvince
      storePostalCode
    }
  }
`;
