// @evershop/evershop/src/modules/catalog/pages/admin/all/CatalogMenuGroup.jsx
import PropTypes from 'prop-types';
import React from 'react';
import AttributeIcon from '@heroicons/react/solid/esm/HashtagIcon';
import CategoryIcon from '@heroicons/react/solid/esm/LinkIcon';
import CollectionIcon from '@heroicons/react/solid/esm/TagIcon';
import ProductIcon from '@heroicons/react/solid/esm/ArchiveIcon';
import NavigationItemGroup from '@components/admin/cms/NavigationItemGroup';
import {useTranslation} from "react-i18next";
import Layout from "@evershop/evershop/src/modules/cms/pages/frontStore/all/Layout";
import '../../../../i18n'


export default function CatalogMenuGroup({

  productGrid,
  categoryGrid,
  attributeGrid,
  collectionGrid
}) {
    const {t, i18n} = useTranslation();
  return (
    <NavigationItemGroup
      id="catalogMenuGroup"
      name= {t('catalog')}
      items={[
        {
          Icon: ProductIcon,
          url: productGrid,
          title: t('products')
        },
        {
          Icon: CategoryIcon,
          url: categoryGrid,
          title: t('categories')
        },
        {
          Icon: CollectionIcon,
          url: collectionGrid,
          title: t('collections')
        },
        {
          Icon: AttributeIcon,
          url: attributeGrid,
          title: t('attributes')
        }
      ]}
    />
  );
}

CatalogMenuGroup.propTypes = {
  attributeGrid: PropTypes.string.isRequired,
  categoryGrid: PropTypes.string.isRequired,
  collectionGrid: PropTypes.string.isRequired,
  productGrid: PropTypes.string.isRequired
};

export const layout = {
  areaId: 'adminMenu',
  sortOrder: 20
};

export const query = `
  query Query {
    productGrid: url(routeId:"productGrid")
    categoryGrid: url(routeId:"categoryGrid")
    attributeGrid: url(routeId:"attributeGrid")
    collectionGrid: url(routeId:"collectionGrid")
  }
`;
