// @evershop/evershop/src/modules/setting/pages/admin/all/SettingMenuGroup.jsx
import PropTypes from 'prop-types';
import React, {useState} from 'react';
import CogIcon from '@heroicons/react/solid/esm/CogIcon';
import NavigationItemGroup from '@components/admin/cms/NavigationItemGroup';
import {useTranslation} from "react-i18next";
import '../../../../i18n'
import Layout from "@evershop/evershop/src/modules/cms/pages/frontStore/all/Layout";

export default function CmsMenuGroup({storeSetting}) {
  const {t, i18n} = useTranslation()



  return (
      <>
        <NavigationItemGroup
            id="settingMenuGroup"
            name={t('settingname')}
            // eslint-disable-next-line react/no-unstable-nested-components
            Icon={() => <CogIcon width={15} height={15}/>}
            url={storeSetting}
        />

      </>
  );
}

CmsMenuGroup.propTypes = {
  storeSetting: PropTypes.string.isRequired
};

export const layout = {
  areaId: 'adminMenu',
  sortOrder: 500
};

export const query = `
  query Query {
    storeSetting: url(routeId:"storeSetting")
  }
`;
